### ISSUE TITLE
[scan] Race condition between simulator reset and force starting

### ISSUE THREAD

**Issue by jaleksynas** [fastlane/fastlane#14717]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
When using the `reset_simulator` option and the hidden `FASTLANE_EXPLICIT_OPEN_SIMULATOR` option, one simulator is started _before_ the reset and therefore is (a) not reset and (b) causes reset to throw an error since it is already booted.

##### Command executed
`export FASTLANE_EXPLICIT_OPEN_SIMULATOR=2 fastlane scan`

##### Abbreviated output when running fastlane
```
15:28:11 Killing all running simulators
15:28:13 Resetting iPhone 8
15:28:14 An error was encountered processing the command (domain=com.apple.CoreSimulator.SimError, code=164):
15:28:14 Unable to erase contents and settings in current state: Booted
```

`Killing all running simulators` - https://github.com/fastlane/fastlane/blob/2326f410e26f37632cef0858305f4e62e47dbd84/scan/lib/scan/runner.rb#L31

Reset & error -
https://github.com/fastlane/fastlane/blob/2326f410e26f37632cef0858305f4e62e47dbd84/scan/lib/scan/runner.rb#L43-L46

#####  Fix Ideas.
1. Make the `reinstall_app` and `reset_simulator` conflicting options, add `force_quit_simulators` option.  And the first three steps in runner are:
```
`killall Simulator &> /dev/null` if Scan.config['force_quit_simulators']

Scan.devices.each do |device|
    FastlaneCore::Simulator.reset(udid: device.udid)
end if Scan.config[:reset_simulator]

open_simulator_for_device(Scan.devices.first) if Scan.devices
```

2. Move the if/elsif block for  `reset_simulator`/`reinstall_app` above the open_simulator_for_device call.

**Comment by janpio** [comment]
The two remaining uses of `FASTLANE_EXPLICIT_OPEN_SIMULATOR` look a bit like artifacts that were forgotten to be removed. Should `FASTLANE_EXPLICIT_OPEN_SIMULATOR` be documented? What do you use it for?

**Comment by jaleksynas** [comment]
Resolved by PR #14718
